src/transformers/generation/logits_process.py

```diff
@@ -1745,8 +1746,14 @@ class ForceTokensLogitsProcessor(LogitsProcessor):
     ```
     """
 
-    def __init__(self, force_token_map: List[List[int]]):
+    def __init__(self, force_token_map: List[List[int]], _has_warned: Optional[bool] = False):
         self.force_token_map = dict(force_token_map)
+        if not _has_warned:
```

Comment:
This argument is to distinguish between when the user has passed `forced_ids` to generate, in which case we throw a warning explaining so:
https://github.com/huggingface/transformers/blob/4a604ed2cc6ed788f5946de20f497cb2dac10d7f/src/transformers/generation/utils.py#L921-L926

OR if the user instantiates the forced logits processor standalone. Again, we give a descriptive warning explaining this:
https://github.com/huggingface/transformers/blob/4a604ed2cc6ed788f5946de20f497cb2dac10d7f/src/transformers/generation/logits_process.py#L1751-L1756

=> we don't want to trigger the second warnings if the user does option 1, since the processor has been instantiated automatically. Thus, we have a separate warning for each case.